### PR TITLE
Remove unnecessary components and dependencies

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -47,7 +52,7 @@
 
     <body>
         <noscript>
-            <%= process.env.MY_APP_NAME %> needs JS.
+            Manager Dashboard needs JS.
```
Comment (tnagorra) [inline app/index.html:55]: You do not need to change this. It's automatically set by webpack.

File: app/index.html [hunk 2]
```diff
@@ -1,41 +1,46 @@
 <!DOCTYPE html>
 <html lang="en" translate="no">
     <head>
-        <title><%= process.env.MY_APP_NAME %></title>
+        <title>MapSwipe Manager Dashboard</title>
```
Comment (tnagorra) [inline app/index.html:4]: You do not need to change this. It's automatically set by webpack.

File: app/components/Modal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                        <h4 className={styles.titleContainer}>
+                            {heading}
+                        </h4>
+                        {!closeButtonHidden && (
+                            <div className={styles.actions}>
+                                <Button
+                                    className={styles.closeButton}
+                                    onClick={onCloseButtonClick}
+                                    transparent
+                                    name="Close"
```
Comment (tnagorra) [inline app/components/Modal/index.tsx:56]: Let's add a title prop

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/Base/configs/firebase.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    projectId: PROJECT_ID,
+    storageBucket: STORAGE_BUCKET,
+    messagingSenderId: MESSAGING_SENDER_ID,
+    appId: APP_ID,
+};
+
+let firebaseApp: FirebaseApp;
+
+export function initFirebaseApp() {
+    firebaseApp = initializeApp(firebaseConfig);
```
Comment (tnagorra) [inline app/Base/configs/firebase.ts:27]: Let's also add a log if we try to init twice.

File: app/Base/components/Navbar/index.tsx [hunk 5]
```diff
@@ -24,87 +18,68 @@ interface Props {
...
-            // FIXME: handle failure
-        },
-    );
-    */
-    const logout = useCallback(
-        () => {
+    const handleLogoutClick = React.useCallback(async () => {
+        const auth = getAuth();
+        try {
+            await auth.signOut();
```
Comment (tnagorra) [inline app/Base/components/Navbar/index.tsx:29]: Let's handle loading state

File: app/views/NewProject/index.tsx [hunk 6]
```diff
@@ -0,0 +1,435 @@
...
+            setFieldValue(projectName, 'name');
+        },
+        [
+            setFieldValue,
+            value?.projectTopic,
+            value?.projectRegion,
+            value?.projectNumber,
+            value?.requestingOrganization,
+        ],
+    );
```
Comment (tnagorra) [inline app/views/NewProject/index.tsx:112]: Can we instead wrap change handlers?

File: app/views/NewProject/index.tsx [hunk 7]
```diff
@@ -0,0 +1,435 @@
...
+                || value.projectType === PROJECT_TYPE_CHANGE_DETECTION) {
+                setFieldValue(25, 'groupSize');
+            }
+
+            if (value.projectType === PROJECT_TYPE_COMPLETENESS) {
+                setFieldValue(80, 'groupSize');
+            }
+        },
+        [setFieldValue, value?.projectType],
+    );
```
Comment (tnagorra) [inline app/views/NewProject/index.tsx:148]: Can we instead wrap change handlers?

File: app/views/NewProject/index.tsx [hunk 8]
```diff
@@ -0,0 +1,435 @@
...
+
+    const handleFormSubmission = React.useCallback((finalValues: PartialProjectFormType) => {
+        console.info(finalValues);
```
Comment (tnagorra) [inline app/views/NewProject/index.tsx:151]: Let's handle this

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by laurentS — COMMENTED** [review]
Can I simply suggest that you setup a very basic CI check (with github actions to stay consistent with the other repos) that does linting and type checking at a minimum? I cannot figure out if you've picked a formatter, but I'd strongly recommend it if not (and check formatting in CI).
In my experience, doing this early helps quite a lot, and is very cheap.
EDIT: ignore me, I just saw the checks below. well done :)

**Comment by tnagorra** [comment]
Hi @laurentS 

Thanks for the reminder about the CI/CD.

We have implemented a simple CI pipeline that includes:
- test runner
- code linting
- build check

You can take a look at the current progress and play around in our test environment at https://mapswipe-manager-dashboard.dev.togglecorp.com. It uses the dev mapswipe firebase instance.

cc: @Hagellach37 @danbjoseph

**Comment by tnagorra** [comment]
Let's also rename "Organization" to "Organisation" through out the project.

File: app/components/FileInput/index.tsx [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+        const handleFileLoad = () => {
+            setImageUrl(String(fileReader.result) ?? undefined);
+        };
+
+        fileReader.addEventListener('load', handleFileLoad);
+        fileReader.readAsDataURL(file);
+
+        return () => {
+            fileReader.removeEventListener('load', handleFileLoad);
+        };
```
Comment (tnagorra) [inline app/components/FileInput/index.tsx:44]: It's better if we could use similar mechanism to GeoJSON Input

File: app/components/FileInput/index.tsx [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+        label,
+        labelContainerClassName,
+        readOnly,
+        accept,
+        showPreview,
+        description,
+        descriptionContainerClassName,
+    } = props;
+
+    const [inputId] = React.useState(randomString());
```
Comment (tnagorra) [inline app/components/FileInput/index.tsx:98]:
```suggestion
    const [inputId] = React.useState(() => randomString());
```

File: app/components/GeoJsonFileInput/index.tsx [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+        }
+        // NOTE: booleanValid does not support FeatureCollection and GeometryCollection
+        // NOTE: booleanValid does seem to support Feature but it does not
+        return booleanValid(value);
+    } catch {
+        return false;
+    }
+}
+
+function readUploadedFileAsText(inputFile: File) {
```
Comment (tnagorra) [inline app/components/GeoJsonFileInput/index.tsx:33]: Let's move this to utils

File: app/components/GeoJsonPreview/index.tsx [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+                    4,
+                );
+
+                const layer = tileLayer(
+                    'https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}.png',
+                    {
+                        attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a>',
+                        subdomains: ['a', 'b', 'c'],
+                    },
+                );
```
Comment (tnagorra) [inline app/components/GeoJsonPreview/index.tsx:46]: Let's check why this does not load in locally hosted server

File: app/hooks/useFirebaseDatabase.tsx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+        const handleQueryError = (error: unknown) => {
+            // eslint-disable-next-line no-console
+            console.error(error);
+            setPending(false);
+        };
+
+        const unsubscribe = onValue(query, handleQueryDone, handleQueryError);
+
+        return unsubscribe;
```
Comment (tnagorra) [inline app/hooks/useFirebaseDatabase.tsx:42]: We may need to include `setPending(false)` when unsubscribing.

File: app/views/UserGroups/index.tsx [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+    const {
+        data: userGroups,
+        pending,
+    } = useFirebaseDatabase<UserGroup>({
+        query: userGroupsQuery,
+    });
+
+    const userGroupList = React.useMemo(
+        () => {
+            const list = Object.entries(userGroups ?? {});
```
Comment (tnagorra) [inline app/views/UserGroups/index.tsx:57]:
```suggestion
            const list = Object.entries(userGroups) ?? [];
```

File: app/views/UserGroups/UserGroupItem/index.tsx [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+
+                    await update(databaseRef(db), updates);
+                    if (!mountedRef.current) {
+                        return;
+                    }
+                    setArchivePending(false);
+                } catch (submissionError) {
+                    // eslint-disable-next-line no-console
+                    console.error(submissionError);
+                    setArchivePending(false);
```
Comment (tnagorra) [inline app/views/UserGroups/UserGroupItem/index.tsx:84]: We need to check if it's mounted here as well. The catch block can run after an async function throws

File: app/views/UserGroups/UserGroupItem/index.tsx [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+
+                                if (index === (memberIdList.length - 1)) {
+                                    setUserListPending(false);
+                                }
+                            },
+                            { onlyOnce: true },
+                        );
+                    },
+                );
+            }
```
Comment (tnagorra) [inline app/views/UserGroups/UserGroupItem/index.tsx:132]: Let's return early.
Why are we iterating on memberIds and adding onValue handlers? Is this to listen for change?

File: app/views/UserGroups/UserGroupFormModal/index.tsx [hunk 17]
```diff
@@ -0,0 +1,254 @@
...
+                            setSubmissionStatus('success');
+                        } else {
+                            setNonFieldError('Failed to push new key for user group');
+                            setSubmissionStatus('failed');
+                        }
+                    },
+                    { onlyOnce: true },
+                );
+            } catch (submissionError) {
+                setSubmissionStatus('failed');
```
Comment (tnagorra) [inline app/views/UserGroups/UserGroupFormModal/index.tsx:132]: We need to check if it's mounted here as well. The catch block can run after an async function throws

File: app/views/UserGroups/UserGroupFormModal/index.tsx [hunk 18]
```diff
@@ -0,0 +1,254 @@
...
+                                createdBy: user?.id,
+                            };
+
+                            const putUserGroupRef = databaseRef(db, `v2/userGroups/${newKey}`);
+                            await setToDatabase(putUserGroupRef, uploadData);
+                            setSubmissionStatus('success');
+                        } else {
+                            setNonFieldError('Failed to push new key for user group');
+                            setSubmissionStatus('failed');
+                        }
```
Comment (tnagorra) [inline app/views/UserGroups/UserGroupFormModal/index.tsx:127]: Let's return early.

File: app/views/UserGroups/UserGroupFormModal/index.tsx [hunk 19]
```diff
@@ -0,0 +1,254 @@
...
+                        if (newKey) {
+                            const uploadData = {
+                                ...finalValues,
+                                nameKey,
+                                createdBy: user?.id,
+                            };
+
+                            const putUserGroupRef = databaseRef(db, `v2/userGroups/${newKey}`);
+                            await setToDatabase(putUserGroupRef, uploadData);
+                            setSubmissionStatus('success');
```
Comment (tnagorra) [inline app/views/UserGroups/UserGroupFormModal/index.tsx:123]: Check if it's mounted after await

File: app/views/UserGroups/UserGroupFormModal/index.tsx [hunk 20]
```diff
@@ -0,0 +1,254 @@
...
+                            setError((prevValue) => ({
+                                ...getErrorObject(prevValue),
+                                name: 'A group with this name already exists, please use a different name (Please note that the name comparision is not case sensitive)',
+                            }));
+                            setSubmissionStatus(undefined);
+                            return;
+                        }
+
+                        const newUserGroupRef = await pushToDatabase(userGroupsRef);
+                        const newKey = newUserGroupRef.key;
```
Comment (tnagorra) [inline app/views/UserGroups/UserGroupFormModal/index.tsx:112]: Check if it's still mounted after await

File: app/views/Teams/index.tsx [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+    );
+
+    const {
+        data: teams,
+        pending,
+    } = useFirebaseDatabase<Team>({
+        query: teamsQuery,
+    });
+
+    const teamList = Object.entries(teams ?? {});
```
Comment (tnagorra) [inline app/views/Teams/index.tsx:46]:
```suggestion
    const teamList = Object.entries(teams) ?? [];
```

File: app/views/Teams/TeamItem/index.tsx [hunk 22]
```diff
@@ -0,0 +1,152 @@
...
+
+    const {
+        data: teamMembers,
+        pending,
+    } = useFirebaseDatabase<User>({
+        skip: !showDetails,
+        query: teamMembersQuery,
+    });
+
+    const teamMemberList = Object.entries(teamMembers ?? {});
```
Comment (tnagorra) [inline app/views/Teams/TeamItem/index.tsx:73]:
```suggestion
    const teamMemberList = teamMembers ? Object.entries(teamMembers) : [];
```

File: app/views/NewProject/utils.ts [hunk 23]
```diff
@@ -0,0 +1,411 @@
...
+    if (
+        isFalsyString(projectTopic)
+        || isNotDefined(projectNumber)
+        || isFalsyString(projectRegion)
+        || isFalsyString(requestingOrganisation)
+    ) {
+        return undefined;
+    }
+
+    return `${projectTopic} - ${projectRegion}(${projectNumber}) ${requestingOrganisation}`;
```
Comment (tnagorra) [inline app/views/NewProject/utils.ts:264]:
```suggestion
    return `${projectTopic} - ${projectRegion} (${projectNumber}) ${requestingOrganisation}`;
```

File: app/views/NewProject/index.tsx [hunk 24]
```diff
@@ -0,0 +1,801 @@
...
+                    setError((err) => ({
+                        ...getErrorObject(err),
+                        TMId: res.error,
+                    }));
+                } else {
+                    setError((err) => ({
+                        ...getErrorObject(err),
+                        TMId: res.message,
+                    }));
+                }
```
Comment (tnagorra) [inline app/views/NewProject/index.tsx:250]: We should use ternary operator instead

File: app/views/NewProject/index.tsx [hunk 25]
```diff
@@ -0,0 +1,801 @@
...
+                    setError((err) => ({
+                        ...getErrorObject(err),
+                        geometry: res.error,
+                    }));
+                } else {
+                    setError((err) => ({
+                        ...getErrorObject(err),
+                        geometry: res.message,
+                    }));
+                }
```
Comment (tnagorra) [inline app/views/NewProject/index.tsx:231]: We should use ternary operator instead

File: app/views/NewProject/index.tsx [hunk 26]
```diff
@@ -0,0 +1,801 @@
...
+                    [internal]: 'Some error occurred',
+                }));
+                setProjectSubmissionStatus('failed');
+            }
+        }
+
+        submitToFirebase();
+    }, [user, mountedRef, setError]);
+
+    /*
```
Comment (tnagorra) [inline app/views/NewProject/index.tsx:380]: Let's remove this

File: app/views/NewProject/index.tsx [hunk 27]
```diff
@@ -0,0 +1,801 @@
...
+                            labelSelector={labelSelector}
+                        />
+                    </div>
+                    <TextInput
+                        name={'name' as const}
+                        value={value?.name}
+                        label="Name"
+                        hint="We will generate you project name based on your inputs above."
+                        readOnly
+                        placeholder="[Project Topic] - [Project Region]([Task Number]) [Requesting Organisation]"
```
Comment (tnagorra) [inline app/views/NewProject/index.tsx:477]:
```suggestion
                        placeholder="[Project Topic] - [Project Region] ([Task Number]) [Requesting Organisation]"
```

File: app/views/Login/index.tsx [hunk 28]
```diff
@@ -0,0 +1,135 @@
...
+
+    const { setUser } = React.useContext(UserContext);
+
+    const [errorMessage, setErrorMessage] = React.useState<string>();
+    const [pending, setPending] = React.useState(false);
+
+    const [email, setEmail] = useInputState<string | undefined>('');
+    const [password, setPassword] = useInputState<string | undefined>('');
+
+    const handleLoginClick = React.useCallback(async () => {
```
Comment (tnagorra) [inline app/views/Login/index.tsx:37]: Lets use useMountedRef here

File: app/Base/components/Navbar/index.tsx [hunk 29]
```diff
@@ -24,87 +18,83 @@ interface Props {
...
 
-    const [
-        modal,
-        onLogoutClick,
-    ] = useConfirmation<undefined>({
-        showConfirmationInitially: false,
-        onConfirm: logout,
-        message: 'Are you sure you want to logout?',
-    });
+    const handleLogoutClick = React.useCallback(async () => {
```
Comment (tnagorra) [inline app/Base/components/Navbar/index.tsx:28]: Let's add useMountedRef

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/components/GeoJsonPreview/index.tsx [hunk 30]
```diff
@@ -0,0 +1,123 @@
...
+    React.useEffect(
+        () => {
+            if (mapContainerRef.current && !mapRef.current) {
+                mapRef.current = createMap(mapContainerRef.current);
+            }
+
+            if (mapRef.current) {
+                mapRef.current.setView(
+                    [0.0, 0.0],
+                    4,
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply app/components/GeoJsonPreview/index.tsx:37]: bump

File: app/components/GeoJsonPreview/index.tsx (hunk 12 above)
Comment (tnagorra) [reply app/components/GeoJsonPreview/index.tsx:47]: bump

File: app/components/GeoJsonPreview/index.tsx [hunk 31]
```diff
@@ -0,0 +1,123 @@
...
+        () => {
+            if (!geoJson) {
+                return undefined;
+            }
+
+            if (!mapRef.current) {
+                return undefined;
+            }
+
+            if (mapRef.current) {
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply app/components/GeoJsonPreview/index.tsx:73]: bump

File: app/components/GeoJsonPreview/index.tsx [hunk 32]
```diff
@@ -0,0 +1,123 @@
...
+                newGeoJson.addData(geoJson);
+                const bounds = newGeoJson.getBounds();
+
+                if (bounds.isValid()) {
+                    mapRef.current.fitBounds(bounds);
+                }
+            }
+
+            return () => {
+                if (geoJsonLayerRef.current && mapRef.current) {
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply app/components/GeoJsonPreview/index.tsx:87]: bump

File: app/components/GeoJsonPreview/index.tsx (hunk 12 above)
Comment (frozenhelium) [reply app/components/GeoJsonPreview/index.tsx:47]: We'll be changing the tile layer, so let's ignore this one for now

File: app/views/UserGroups/UserGroupItem/index.tsx (hunk 16 above)
Comment (frozenhelium) [reply app/views/UserGroups/UserGroupItem/index.tsx:132]: Fetching details for all users that belong in the group, there isn't any better way :/

**Pull request merged**